test/canillita_newsitems_SUITE.erl

```diff
@@ -0,0 +1,219 @@
+  ct:comment("content-type must be provided for POST"),
+  #{status_code := 415} =
+    canillita_test_utils:api_call(post, NewsItem1Url, NoHeaders, <<>>),
+
+  ct:comment("content-type must be JSON for POST"),
+  #{status_code := 415} =
+    canillita_test_utils:api_call(post, NewsItem1Url, InvalidHeaders, <<>>),
+
+  ct:comment("Agent must accept JSON for POST"),
+  #{status_code := 406} =
+    canillita_test_utils:api_call(post, NewsItem1Url, InvalidAccept, <<>>),
+
+  {comment, ""}.
+
+-spec invalid_parameters(Config::canillita_test_utils:config()) ->
+  {comment, string()}.
+invalid_parameters(_Config) ->
+  Headers = #{<<"content-type">> => <<"application/json">>},
+  Newspaper1 = create_newspaper(<<"newspaper1">>, <<"description1">>),
+  NewsItem1Url = "/newspapers/"++Newspaper1++"/news",
+
+  ct:comment("Empty or broken parameters are reported"),
+  #{status_code := 400} =
+    canillita_test_utils:api_call(post, NewsItem1Url, Headers, <<>>),
+  #{status_code := 400} =
+    canillita_test_utils:api_call(post, NewsItem1Url, Headers, <<"{">>),
+
+  ct:comment("Missing parameters are reported"),
+  None = #{},
+  #{status_code := 400} =
+    canillita_test_utils:api_call(post, NewsItem1Url, Headers, None),
+
+  NoTitle = #{body => <<"notitle">>},
+  #{status_code := 400} =
+    canillita_test_utils:api_call(post, NewsItem1Url, Headers, NoTitle),
+
+  NoBody = #{title => <<"nobody">>},
+  #{status_code := 400} =
+    canillita_test_utils:api_call(post, NewsItem1Url, Headers, NoBody),
+
+  {comment, ""}.
+
+-spec not_found(Config::canillita_test_utils:config()) -> {comment, string()}.
+not_found(_Config) ->
+  Headers = #{<<"content-type">> => <<"application/json; charset=utf-8">>},
+  NewsItem = #{title => <<"titlex">>, body => <<"bodyx">>},
+  ct:comment("Unable to create a newsitem with a non-existing newspaper"),
+  #{status_code := 404} =
+    canillita_test_utils:api_call( post
+                                 , "/newspapers/non-existing/news"
+                                 , Headers
+                                 , NewsItem
+                                 ),
+
+  Newspaper1 = create_newspaper(<<"newspaper1">>, <<"description1">>),
+  Newspaper1Bin = list_to_binary(Newspaper1),
+  NewsItem1Url = "/newspapers/"++Newspaper1++"/news",
+
+  ct:comment("Create a newsitem"),
+  #{status_code := 201, body := Body1} =
+    canillita_test_utils:api_call(
+      post
+    , "/newspapers/"++Newspaper1++"/news"
+    , Headers
+    , #{ <<"title">> => <<"title1">>
+       , <<"body">> => <<"body1">>
+       }
+    ),
+  #{ <<"id">> := NewsItem1Id
+   , <<"newspaper_name">> := Newspaper1Bin
+   , <<"title">> := <<"title1">>
+   , <<"body">> := <<"body1">>
+   , <<"created_at">> := _CreatedAt
+   } = sr_json:decode(Body1),
+
+  ct:comment("Unable to get non-existing newsitem with a valid newspaper"),
+  #{status_code := 404} =
+    canillita_test_utils:api_call(get, NewsItem1Url++"/non-existing-id"),
```

Comment:
I'm not sure why @gadgetci has not complained about the lack of spaces around `++`, but it should've complained